### DISCUSSION TITLE
Increased program graphical customization

### DISCUSSION THREAD

**Discussion by goliv04053** [nicotine-plus/nicotine-plus#2277] [Ideas]
### Describe the solution you'd like
More customization on theme, colors, fonts, skins, background transparency and etc.

### Additional context
An example that could be an inspiration would be Winamp with its various skins and even the customization available in KDE plasma.

**Comment by mathiascode** [comment]
Fonts - we have options to set the global font and fonts for various parts of the UI. Is anything lacking here?

Theme, colors, skins, background transparency - larger customizations here aren't maintainable in the long run. GTK themes are supposed to provide most of the customization on Linux distros, and any larger customizations we provide will conflict with these themes. Skins can make sense in a somewhat small application like Winamp, but GTK themes and widgetry are rather complex, and we can't guarantee forward compatibility with major GTK releases.

We could probably provide an option to use a specific GTK theme in Nicotine+.

**Comment by slook** [comment]
> We could probably provide an option to use a specific GTK theme in Nicotine+.

This would be very useful for many users, because on certain systems there is no out-of-the-box method of choosing Gtk themes, Windows for example.

> Theme, colors, skins, background transparency

The low-hanging fruit here would be an option to set a custom background color ListView and TextView, as we previously deliberated in https://github.com/nicotine-plus/nicotine-plus/discussions/1526#discussioncomment-1334379 . This would have a large impact on the overall appearance of the application, that many users who are stuck with a default Gtk theme would appreciate.

> any larger customizations we provide will conflict with these themes

With the default N+ settings, on certain dark Gtk themes, the text color is invisible or inappropriate unless the customizations are used to workaround these broken themes.

**Comment by mathiascode** [comment]
> With the default N+ settings, on certain dark Gtk themes, the text color is invisible or inappropriate unless the customizations are used to workaround these broken themes.

This is no longer the case with newer Nicotine+ defaults since 2020. If the theme itself is broken, it should be reported upstream, since it would surely be unusable everywhere else too (or ask the distro to ship a sensible theme by default). We use standard widgets and practices when possible.

Exactly this is my main concern with fine grained customization and many non-standard theme colors used by default. Themes may evolve over time, or the user may change their theme later, forgetting their custom color choices in the past and wondering why the experience is broken.

**Comment by slook** [comment]
> the user may change their theme later, forgetting their custom color choices in the past and wondering why the experience is broken.

That is fair to say. There are many cases where over-riding colors is a bad idea, such as where a system has dark/light theme variants set depending upon the time of day, for example.

> We could probably provide an option to use a specific GTK theme in Nicotine+.

If this were to become the case, then I suppose the user could acquire or craft any GTK compatible theme of their choosing to use specially for Nicotine+, without breaking the look of other apps on their system. This could be quite a flexible arrangement.

----

There is a somewhat interesting thread about packaging themes for use with a Gtk3 based IDE app for Windows and other platforms https://github.com/geany/geany/issues/3063 including relevant information about folder paths (a somewhat complex issue it seems).

They are in the process of writting a FAQ guide to changing Gtk theme for different OS's:
https://github.com/geany/www.geany.org/blob/8f469954d7d1ee3c288c9bbf2c2d16368d724b91/page_content/documentation/faq.md#how-to-change-the-gtk-theme

----

Gtk apparently provides an easy method of adding or bundling themes with applications, see link:

https://www.gtk.org/docs/installations/windows/#building-and-distributing-your-application

_"Step 4. To make GTK pick up this theme, put a file `settings.ini` in `etc/gtk-3.0` in your installation folder. This should contain_
```
[Settings]
gtk-theme-name=ExampleThemeName
gtk-font-name=Segoe UI 9
```
"

**Comment by goliv04053** [comment]
What about that new GNOME accents colors? That can be one path to go and for me... that's okay.
https://news.itsfoss.com/gnome-accent-colors/
